### PR TITLE
psbt: handle unspendable psbts

### PR DESCRIPTION
**Pull request by achow101**
When analyzing an unspendable PSBT, report that it is unspendable and exit analysis early.

### CODE REVIEW

**Comment by practicalswift** [comment]
Concept ACK

Thanks for fixing this!

**Comment by Sjors** [comment]
Concept ACK

**Comment by practicalswift** [comment]
This is the issue (https://github.com/bitcoin/bitcoin/issues/17149#issuecomment-552202472) fixed by this PR:

```
$ bitcoind &
$ bitcoin-cli analyzepsbt "cHNidP8BAJoCAAAAAljoeiG1ba8MI76OcHBFbDNvfLqlyHV5JPVFiHuyq911AAAAAAD/////g40EJ9DsZQpoqka7CwmK6kQiwHGyyng1Kgd5WdB86h0BAAAAAP////8CcKrwCAAAAAAWAEHYXCtx0AYLCcmIauuBXlCZHdoSTQDh9QUAAAAAFv8/wADXYP/7//////8JxOh0LR2HAI8AAAAAAAEBIADC6wsAAAAAF2oUt/X69ELjeX2nTof+fZ10l+OyAokDAQcJAwEHEAABAACAAAEBIADC6wsAAAAAF2oUt/X69ELjeX2nTof+fZ10l+OyAokDAQcJAwEHENkMak8AAAAA"
bitcoind: consensus/tx_verify.cpp:130: unsigned int 
    GetP2SHSigOpCount(const CTransaction &, const CCoinsViewCache &): 
    Assertion `!coin.IsSpent()' failed.
```

File: test/functional/rpc_psbt.py [hunk 1]
```diff
@@ -416,5 +416,10 @@ def test_psbt_input_keys(psbt_input, keys):
         analyzed = self.nodes[0].analyzepsbt(signed)
         assert analyzed['inputs'][0]['has_utxo'] and analyzed['inputs'][0]['is_final'] and analyzed['next'] == 'extractor'
 
+        self.log.info("PSBT spending unspendable outputs should have error message and Creator as next")
+        analysis = self.nodes[0].analyzepsbt('cHNidP8BAJoCAAAAAljoeiG1ba8MI76OcHBFbDNvfLqlyHV5JPVFiHuyq911AAAAAAD/////g40EJ9DsZQpoqka7CwmK6kQiwHGyyng1Kgd5WdB86h0BAAAAAP////8CcKrwCAAAAAAWAEHYXCtx0AYLCcmIauuBXlCZHdoSTQDh9QUAAAAAFv8/wADXYP/7//////8JxOh0LR2HAI8AAAAAAAEBIADC6wsAAAAAF2oUt/X69ELjeX2nTof+fZ10l+OyAokDAQcJAwEHEAABAACAAAEBIADC6wsAAAAAF2oUt/X69ELjeX2nTof+fZ10l+OyAokDAQcJAwEHENkMak8AAAAA')
```
Comment (instagibbs) [inline test/functional/rpc_psbt.py]: damnit you should have warned me this crashes Core, lol

**Comment by instagibbs** [comment]
@practicalswift 2 minutes too late bro :joy:

File: test/functional/rpc_psbt.py (hunk 1 above)
Comment (instagibbs) [reply test/functional/rpc_psbt.py]:
```
      "final_scriptSig": {
        "asm": "1050369 0 0 0 OP_LEFT",
        "hex": "030107100001000080"
      }
```

why does this PSBT have scriptsigs for OP_RETURN inputs? Can't this PSBT be generated dynamically for better readability? Even if not, we should assert some things using `decodepsbt` to make sure we know the scriptpubkeys.

**Comment by practicalswift** [comment]
@instagibbs 

<details>
<summary>Trigger warning: old meme</summary>

image

</details>

File: test/functional/rpc_psbt.py (hunk 1 above)
Comment (practicalswift) [reply test/functional/rpc_psbt.py]: The PSBT was created by a coverage-guided fuzzer using the fuzzing harness that I added in PR #17136 ("tests: Add fuzzing harness for various PSBT related functions"). The only post-processing done was a test-case minimisation run :)

**Comment by Sjors** [comment]
ACK 773d457

The new `creator` role and `error` field are release note worthy.

**Comment by darosior** [comment]
The removal of the input field in case of error would break the API, wouldn't be better to push an empty array instead ?

**Comment by achow101** [comment]
I don't think this really breaks the api since this error condition would cause a crash rather than return any sort of result previously.

**Comment by instagibbs** [comment]
After some thought ACK https://github.com/bitcoin/bitcoin/commit/773d4572a4864ab7b6380858d07d9579ff6dd9a2

**Pull request merged**